wasm/src/program/transaction.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{
+  account::{Address, PrivateKey},
+  record::RecordPlaintext
+};
+use crate::{
+    Aleo,
+    Identifier,
+    Process,
+    Program,
+    ProvingKey,
+    TransactionNative
+};
+
+use std::str::FromStr;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct TransactionBuilder {}
+
+#[wasm_bindgen]
+impl TransactionBuilder {
+    /// Creates an execute transaction from a full proof of execution
+    pub fn build_transfer_full(
+        private_key: PrivateKey,
+        proving_key: ProvingKey,
+        address: Address,
+        amount: u64,
+        record: RecordPlaintext,
```

Comment:
I think this at its core is the right abstraction, passing the proving key. We will probably also need the inclusion proof data here too.

Below suggest just passing the PrivateKey as you can derive the record from it and renaming amount to "gates" to disambiguate them from credits.
```suggestion
        private_key: PrivateKey,
        proving_key: ProvingKey,
        gates: u64,
        record: RecordPlaintext,
```

Reply:
I'm not quite sure what you're suggesting here. Are you suggesting we pass in an `RecordCiphertext` instead of a `RecordPlaintext` b/c we also have to use the private key?

Reply:
Oh yes, I'm sorry, I meant 'gates' not records!